ducanhpham0312/zeevision
# feat: fetch api for processes

### Linked issue

Closes ducanhpham0312/zeevision-private#13


### Description

Added endpoint for fetching processes and single process from database through GraphQL API.

### Testing

1. Start test bench
2. `docker compose up --build`
3. Navigate to processes page in ZeeVision; the table should be empty
4. Follow backend `README` to open pgAdmin (NOTE: usernames and password have been changed/simplified)
5. Open `Servers->zeevision_db->Databases->zeevision_db` which will enable a query tool. Open the query tool:

![image](https://github.com/ducanhpham0312/zeevision/assets/11910622/b150cec1-46bc-4d33-ae2f-acb4410653f6)

6. Write the following query to the query field and run it with the play button (`Execute/Refresh`):

```sql
INSERT INTO processes(id, process_id, process_key, bpmn_resource)
VALUES(1, 1, 1000, 'BPMN file if we had one');
```

7. Open ZeeVision processes page again; there should be the record you just inserted. You can insert multiple records to test out the feature.
